coala_quickstart/info_extractors/GruntfileInfoExtractor.py

```diff
@@ -0,0 +1,265 @@
+        """
+        search_results = search_object_recursively(
+            parsed_file, "callee",
+            {
+                "computed": False,
+                "type": "MemberExpression",
+                "property": {
+                    "name": "loadNpmTasks",
+                    "type": "Identifier"
+                    },
+                "object": {
+                    "name": "grunt",
+                    "type": "Identifier"
+                    }
+            })
+
+        tasks_found = []
+
+        for result in search_results:
+            if result.get("object"):
+                task_name = result["object"]["arguments"][0]["value"]
+                tasks_found.append(task_name)
+
+        return tasks_found
+
+    def extract_literals_from_expression(self, obj):
+        """
+        From the given expression of PyJsParser.parse() instance,
+        return the elements of type "Literal" by searching recursively.
+        """
+        results = []
+        if (obj and obj.get("type") == "ArrayExpression" and
+                "elements" in obj.keys()):
+            results = [
+                elem["value"]
+                for elem in obj["elements"] if elem["type"] == "Literal"
+            ]
+
+        elif obj and obj.get("type") == "ObjectExpression":
+            if (obj.get("value") and
+                obj["value"]["type"] == "ArrayExpression" and
+                    "elements" in obj["value"].keys()):
+                results = [
+                    elem["value"]
+                    for elem in obj["elements"] if elem["type"] == "Literal"
+                ]
+            else:
+                return self.extract_literals_from_expression(obj.get("value"))
+
+        elif obj and obj.get("type") == "Property":
+            return self.extract_literals_from_expression(obj.get("value"))
+
+        return results
+
+    def get_task_config(self, parsed_config):
+        """
+        From the parsed configuration object for a task,
+        extract and return the config settings in the form
+        of a python dict.
+        """
+        result = {}
+        for prop in parsed_config["properties"]:
+            prop_value = None
+            if prop["value"]["type"] == "Identifier":
+                prop_value = prop["value"]["name"]
+
+            elif prop["value"]["type"] == "Literal":
+                prop_value = prop["value"]["value"]
+
+            elif prop["value"]["type"] == "ObjectExpression":
+                prop_value = self.get_task_config(prop["value"])
+
+            elif prop["value"]["type"] == "ArrayExpression":
+                prop_value = self.extract_literals_from_expression(prop)
+
+            else:
+                logging.warn(
+                    "quickstart was not able to parse the value "
+                    "of the config {}".format(
+                        prop["key"]["name"]))
+
+            result[prop["key"]["name"]] = prop_value
```

Comment:
If it's none we still return it because although we were not able to find the value of the configuration key (most probably it didn't exist),  at least we detected the key. The `None` values can be handled at later stage.